File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution{
    public int solution(int n, int a, int b){
        int answer = 1;
        // 대결할때까지 각각의 번호를 줄임
        while(true){
            // 작은 수가 홀수이고, 차이가 1이여야 대결함
            if(Math.min(a,b) % 2 == 1 && Math.abs(a-b) == 1) break;
            a -= a/2;
            b -= b/2;
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.js
function solution(n,a,b){
    var answer = 0;
    
    while(n >= 2){
        if((a > n / 2 && b > n / 2)||(a <= n / 2 && b <= n / 2)){
            n /= 2;
            if(a > n){
                a -= n;
                b -= n;
            }
        }
        else{
            answer = powNum(n);
            break;
        } 
    }
    if(n === 1) answer = 1;
    

    return answer;
}

function powNum(n){
    let result = 0;
    while(n > 1){
        n = n / 2;
        result++;
    }
    return result;
}